documentation/images/plantuml/agent/state_machines/backhaul_manager_fsm.puml

```diff
@@ -0,0 +1,84 @@
+@startuml backahul_manager_fsm
+
+[*] --> INIT
+INIT --> WAIT_ENABLE
+WAIT_ENABLE --> ENABLED
+ENABLED --> MASTER_DISCOVERY : wired
+ENABLED --> WIRELESS_FSM : wireless
+state WIRELESS_FSM {
+    [*] --> INIT_HAL
+    INIT_HAL --> WPA_ATTACH
+    WPA_ATTACH --> CONNECT
+    CONNECT --> MASTER_DISCOVERY : connection success
+}
+MASTER_DISCOVERY --> OPERATIONAL
+OPERATIONAL --> RESTART : wired link status change
+RESTART --> INIT
+
+@enduml
+
+@startuml dynamic_backhaul_switch
+
+box "controller" #LightSlateGray
+participant controller
+end box
+
+box "prplmesh agent" #LightBlue
+participant backhaul_manager
+participant radio_agent
+end box
+
+box "platoform"
+participant hostapd
+participant supplicant
+end box
+
+== platform init ==
+hostapd <- : start disabled (static channel)
+activate hostapd
+deactivate hostapd
+supplicant <- : start (no network config)
+activate supplicant
+deactivate supplicant
+
+== init ==
+...
+== operational ==
+controller -> backhaul_manager : switch_backhaul(backhaul)
+activate backhaul_manager
+backhaul_manager -> backhaul_manager : restart
+note left
+m_preferred_backhaul = backhaul
+end note
+backhaul_manager -> supplicant : tear down
+note right
+disable all supplicants
+disable wired interface
+end note
+backhaul_manager -> radio_agent: disconnect
```

Comment:
As I wrote: this will no longer happen in unified agent, I think.

Reply:
@morantr please confirm

Reply:
https://github.com/prplfoundation/prplMesh/pull/1423#discussion_r443345749